### ISSUE TITLE
Docs: prompt_lookup.py docstring still references pre-#26 design ("Phase 2 future", "blocked by mlx-lm")

### ISSUE THREAD

**Issue by st-adam** [jjang-ai/vmlx#137]
## Issue

`vmlx_engine/prompt_lookup.py:13-21` says:

```python
Phase 2 (future): actual acceleration via batched verification.
  - The scheduler already holds self.model and can call it directly
    (see _prefill_for_prompt_only_cache for the pattern).
  - Verification: self.model(mx.array([[last_tok, d1, ..., dK]]), cache=...)
    returns K+1 logits in ONE forward pass instead of K sequential passes.
  - KV cache rollback: truncate accepted-prefix KV tensors (same pattern
    as _truncate_cache_to_prompt_length).
  - Blocked today by: BatchGenerator holds KV cache internally; we can't
    extract it mid-stream without upstream mlx-lm changes.
```

This is incorrect post-#26 — Phase 2 (and Phase 3, and the Phase 5 optimization journey) shipped, the BatchGenerator KV-extract path was solved vmlx-side, and the "blocked by mlx-lm" claim is no longer accurate.

## Fix

Update lines 13–21 to describe the shipped state (Phase 2/3/5 done, K=2 for hybrid, adaptive auto-tune, d0 pre-check, scalar logprob acceptance, O(1) n-gram index) and link to #26 description for the optimization journey.

## Why this matters

The stale comment is currently misleading downstream contributors looking at PLD perf opportunities (e.g. the SSM-checkpoint companion issue) into believing Phase 2 is unstarted.

**Comment by jjang-ai** [comment]
Tracked and corrected in the current source branch. The stale `prompt_lookup.py` docstring no longer says PLD acceleration is a future Phase 2 or blocked by mlx-lm; it now documents the split between model-agnostic draft proposal/telemetry and scheduler-owned verification/reinsert/cache restore.

Verification:
- `.venv/bin/python -m pytest -q tests/test_engine_audit.py::TestPromptLookupDocumentation tests/test_vl_video_regression.py::TestVmlx92PldNonMllmGuard tests/test_vl_video_regression.py::TestVmlx92PldGuardOnNonMllm` -> 7 passed
- `git diff --check` -> clean

Leaving this open until the branch is pushed/packaged.

**Comment by jjang-ai** [comment]
Update: the branch work referenced above is now pushed to main in jjang-ai/vmlx@2ec15696.

The prompt-lookup documentation cleanup is no longer local-only. The source now describes the current scheduler-owned draft/verify/reinsert/cache-restore design instead of the old pre-#26 future-work text.

Verification already run before push:
- `tests/test_engine_audit.py::TestPromptLookupDocumentation`
- PLD non-MLLM guard tests in `tests/test_vl_video_regression.py`

Leaving this open only until this source is included in the next packaged release/docs bundle.

**Comment by jjang-ai** [comment]
Closing as fixed in current main.\n\nCurrent source verification:\n- vmlx_engine/prompt_lookup.py no longer describes PLD as future Phase 2 work or blocked by mlx-lm. It now documents the split correctly: prompt_lookup owns model-agnostic draft proposal/telemetry, while the scheduler owns batched verification, cache trim/restore, and request reinsertion.\n- Regression command run fresh: uv run pytest -q tests/test_engine_audit.py::TestPromptLookupDocumentation tests/test_vl_video_regression.py::TestVmlx92PldNonMllmGuard tests/test_vl_video_regression.py::TestVmlx92PldGuardOnNonMllm -> 7 passed.\n\nBoundary: this closes the stale-doc bug. It does not claim new PLD feature work for #134/#135/#136.
